core/embed/rust/src/ui/model_mercury/component/vertical_menu.rs

```diff
@@ -0,0 +1,154 @@
+use heapless::Vec;
+
+use super::theme;
+use crate::ui::{
+    component::{base::Component, Event, EventCtx},
+    display::Icon,
+    geometry::Rect,
+    model_mercury::component::button::{Button, ButtonMsg, IconText},
+    shape::{Bar, Renderer},
+};
+
+pub enum VerticalMenuChoiceMsg {
+    Selected(usize),
+}
+
+/// Number of buttons.
+/// Presently, VerticalMenu holds only fixed number of buttons.
+/// TODO: for scrollable menu, the implementation must change.
+const N_ITEMS: usize = 3;
+
+/// Number of visual separators between buttons.
+const N_SEPS: usize = N_ITEMS - 1;
+
+/// Fixed height of each menu button.
+const MENU_BUTTON_HEIGHT: i16 = 64;
+
+/// Fixed height of a separator.
+const MENU_SEP_HEIGHT: i16 = 2;
+
+type VerticalMenuButtons<T> = Vec<Button<T>, N_ITEMS>;
+type AreasForSeparators = Vec<Rect, N_SEPS>;
+
+pub struct VerticalMenu<T> {
+    area: Rect,
+    /// buttons placed vertically from top to bottom
+    buttons: VerticalMenuButtons<T>,
+    /// areas for visual separators between buttons
+    areas_sep: AreasForSeparators,
+}
+
+impl<T> VerticalMenu<T>
+where
+    T: AsRef<str>,
+{
+    fn new(buttons: VerticalMenuButtons<T>) -> Self {
+        Self {
+            area: Rect::zero(),
+            buttons,
+            areas_sep: AreasForSeparators::new(),
+        }
+    }
+    pub fn select_word(words: [T; 3]) -> Self {
+        let mut buttons_vec = VerticalMenuButtons::new();
+        for (i, word) in words.into_iter().enumerate() {
```

Comment:
Enumerate not needed?

Reply:
https://github.com/trezor/trezor-firmware/pull/3599/commits/5664655e7aaf7b92db44c1e60bac06e9d3b56df0